CMakeLists.txt

```diff
@@ -150,7 +157,12 @@ if(${MM_MALLOC})
 endif()
 
 if(UNIX OR MINGW OR CYGWIN)
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -pthread -O3 -Wextra -Wall -Wno-ignored-attributes -Wno-unknown-pragmas -Wno-return-type")
+    if(USE_BOOST_THREAD)
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1y")
```

Comment:
https://stackoverflow.com/questions/28825113/difference-between-gcc-compile-options-std-c1y-and-std-c14